kafka/producer.go

```diff
@@ -71,7 +80,9 @@ func (ams *asyncMessageSink) doPublishMessages(ctx context.Context, producer sar
 
 	go func() {
 		for suc := range successes {
-			acks <- suc.Metadata.(substrate.Message)
+			msg := suc.Metadata.(substrate.Message)
+			acks <- msg
```

Comment:
Actually just noticed it, not related to your change, but shouldn't this be in a select with the `ctx.Done()` as everything else, as otherwise we might get a goroutine leak?

Reply:
:+1:  I noticed that too actually. I may follow up in another PR, possibly make it user errgroup.